docs/typescript.md

```diff
@@ -131,33 +169,36 @@ const Component: SFC = ({ label, className }) => (
 
 type StyledComponentProps = {
   bgColor: string
-} & ComponentProps
-//  ^^^ You will need this
+};
 
-const StyledComponent =
-  styled <
-  StyledComponentProps >
-  Component`
+const StyledComponent0 = styled(Component)`
   color: red;
-  background: ${props => props.bgColor};
+  background: ${(props: StyledComponentProps) => props.bgColor};
 `
 
+const StyledComponent1 = styled(Component)<StyledComponentProps>({
+  color: 'red',
+}, props => ({
+  background: props.bgColor,
+}));
+
 const App = () => (
-  <StyledComponent bgColor="red" label="Oh, needs to re-type label prop =(" />
+  <div>
+    <StyledComponent0 bgColor="red" label="Oh, needs to re-type label prop =(" />
+    <StyledComponent1 bgColor="red" label="Oh, needs to re-type label prop =(" />
+  </div>
 )
 ```
 
-Unfortunately, when you pass custom props to a styled component, TypeScript will stop inferring your Component props, and you will need to re-type them.
-
 ### Define a Theme
 
 By default, the `props.theme` has `any` type annotation and works without error.\
 However, you can define a theme type by creating a another `styled` instance.
 
-_styled.tsx_
+_styled.jsx_
```

Comment:
Why the change in extension, isn't `tsx` generally used?

Reply:
Oh, my mistake. Sorry. I will change these.